pygount/command.py

```diff
@@ -334,6 +336,9 @@ def execute(self):
 
         try:
             writer_class = _OUTPUT_FORMAT_TO_WRITER_CLASS_MAP[self.output_format]
+
+            if issubclass(writer_class, pygount.write.SummaryWriter):
+                source_paths_and_groups_to_analyze = track(source_paths_and_groups_to_analyze)
```

Comment:
If I read [the progress documentation](https://rich.readthedocs.io/en/stable/progress.html) correctly, this tracks progress by number of files processed.

I suppose it's fine for an initial implementation. Let me know if I should look into making a more ambitious tracking based on file size (after a merge).

Reply:
We should do a small test that times the original execution vs execution with prior file size check. It felt to my like an extremely expensive operation but after using pygount for a while, I have realized just how slow pygments can get so we can try implementing file size tracking even before the merge. I will work on this tomorrow.

Reply:
I've been thinking a bit more about this and I believe for the time being we should leave it at the count based progress.

The main reason being files that can be rather huge but processed very quickly, especially generated and duplicate files. For a decent progress, there would probably need to be two passes:
1. Detect and process files that do not have to be lexed by pygments
2. Process files that are lexed by pygments.

For a total progress, these two passes would need to be accounted for by some heuristic, like pass 1 = 10% and pass 2 = 90%. And the progress within pass 1 would be count based, within pass 2 size based.

But I'd rather we move this two a separate ticked I and look at it on my own (unless you really want to).